fish-rust/src/event.rs

```diff
@@ -201,13 +201,15 @@ pub struct EventDescription {
     typ: EventType,
 }
 
-impl From<&event_description_t> for EventDescription {
-    fn from(desc: &event_description_t) -> Self {
-        EventDescription {
+impl TryFrom<&event_description_t> for EventDescription {
```

Comment:
There's no reason to make this fallible; nobody is prepared to handle this error. If you want to add additional logic such that `--on-signal 0` reports an error to the user then that's OK, but we shouldn't complicate this interface for that silly case.

Reply:
That was the scenario I had in mind, yes. I figured it would be safer to handle it centrally by returning an error rather than require all callers to assert the non-zero value but since there aren't many I suppose that is easier.